### DISCUSSION TITLE
Recommended path for exporting perspective-viewer to PDF

### DISCUSSION THREAD

**Discussion by davisford** [perspective-dev/perspective#2770] [Q&A]
Hello, wondering if the community could help advise on the best recommended path for exporting a perspective-viewer to a PDF so it can be printed or emailed.

If I export to html, it won't print or export beyond a single viewable page because of the shadow dom (doesn't render the rest until scrolling is initiated).

Exporting to csv and then to xlsx then to pdf is horrible.

Right now, I just send the arrow binary back from the server to the client.  I was hoping for a client side solution, but it seems there are really only two valid pathways here:

1. Do it on the server in python, take arrow - create table in perspective-python with our own layout and figure out how to convert that into a pdf
2. Build some kind of WASM plugin that can do this

Is there another recommended way here?  Has anyone else solved this and how did you do it?

**Comment by oleksify** [comment]
@davisford I posted a comment [here](https://github.com/finos/perspective/issues/2772#issuecomment-2379321635), regarding a way I implemented this. I can't post code, but I can help you understand the idea how this can be implemented.

@texodus Sorry, if my comment was misleading or rude. I was trying to reply to original post user. I wasn't trying to ask for any feature, nor complaining something is missing. If it felt so, I'm really sorry.

**Comment by texodus** [comment]
@oleksify You're good, I was responding to a now-deleted ad hominem comment.

Re: PDF - we're not planning on open-sourcing this feature for Perspective.

**Comment by texodus** [comment]
Locking this convo - I'm drawing the line here at advertising proprietary products.
